src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -654,25 +651,12 @@ int calcNIndelInformativeReads(final ReadPileup pileup, final int pileupOffsetIn
      */
     @VisibleForTesting
     protected int getCigarModifiedOffset (final PileupElement p){
-        int offset = p.getOffset();  //index into read base byte[] corresponding to this position in the reference
         final GATKRead read = p.getRead();
-        final Cigar cigar = read.getCigar();
-        final int indexInCigar = p.getCurrentCigarOffset();
-        final CigarElement ce = p.getCurrentCigarElement();
-        final int indexInElement = p.getOffsetInCurrentCigar();
-        for (int i = 0; i < indexInCigar; i++) {
-            if (cigar.getCigarElement(i).getOperator().equals(CigarOperator.INSERTION)) {
-                offset -= cigar.getCigarElement(i).getLength();
+        int offset = p.getCurrentCigarElement().getOperator().consumesReferenceBases() ? p.getOffsetInCurrentCigar() : 0;
+        for (CigarElement elem : read.getCigar().getCigarElements().subList(0, p.getCurrentCigarOffset())) {
```

Comment:
Sorry, I didn't pay close attention that my suggested code was totally correct... the ```int offset = ``` condition ... do you need to add ```|| elem.getOperator() == CigarOperator.S)```?

Reply:
Make ```elem``` ```final```.